core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -961,11 +1012,11 @@ private void clearTenantCache(int tenantId) throws UserStoreException {
         tenantCacheManager.clearCacheEntry(new TenantIdKey(tenantId));
     }
 
-    private void clearTenantCaches(String tenantUniqueID) throws UserStoreException {
+    private void clearTenantCaches(Tenant tenant) throws UserStoreException {
```

Comment:
Why is this changed?

Reply:
Still uses only the tenantUniqueID?

Reply:
In deleteTenant we need tenant domain to call invalidateCacheManager. Therefore getting the tenant before clearTenantCaches will stop calling **this.getTenant(tenantUniqueID)** two times. Thatswhy remove **this.getTenant(tenantUniqueID)** from this method (**clearTenantCaches**) and called it inside deleteTenant(String tenantUniqueID, boolean removeFromPersistentStorage).

Reply:
Previously we have got the whole tenant object using this.getTenant(tenantUniqueID) inside the clearTenantCaches method since we need to get tenantId, tenant domain.

Reply:
Then you can have both clearTenantCaches(String tenantUniqueID) and clearTenantCaches(Tenant tenant).

Reply:
Yes, we can have both. Isn't having two methods with similar functionalities will make duplicate logic? Is there any specific reason to have separate methods as clearTenantCaches(String tenantUniqueID) and clearTenantCaches(Tenant tenant) rather than having one method clearTenantCaches(Tenant tenant) and handled it as I have done currently.

Reply:
The **tenantUniqueID** is the key. We should be able to clear the caches using the key. Otherwise, in each place, we have to getTenant and call the clear method. We should think of code about readability as well. 


Ex. 
public void activateTenant(String tenantUniqueID) throws UserStoreException {
        Tenant tenant = this.getTenant(tenantUniqueID);
        clearTenantCaches(tenant);

public void activateTenant(String tenantUniqueID) throws UserStoreException {
        clearTenantCaches(tenantUniqueID);